src/rpc/blockchain.cpp

```diff
@@ -1532,6 +1539,262 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
     return ret;
 }
 
+static void RpcGetTx(const uint256& hash, CTransactionRef& tx_out)
+{
+    uint256 hashBlock;
+    if (!GetTransaction(hash, tx_out, Params().GetConsensus(), hashBlock, true)) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, std::string(fTxIndex ? "No such mempool or blockchain transaction"
+          : "No such mempool transaction. Use -txindex to enable blockchain transaction queries"));
+    }
+}
+
+template<typename T>
+static T CalculateTruncatedMedian(std::vector<T>& scores)
+{
+    T median;
+    size_t size = scores.size();
+    std::sort(scores.begin(), scores.end());
+    if (size == 0) {
+        median = 0;
+    } else if (size == 1) {
+        median = scores[0];
+    } else if (size % 2 == 0) {
+        median = (scores[size / 2 - 1] + scores[size / 2]) / 2;
+    } else {
+        median = scores[size / 2];
+    }
+    return median;
+}
+
+// outpoint (needed for the utxo index) + nHeight + fCoinBase
+static const size_t PER_UTXO_OVERHEAD = sizeof(COutPoint) + sizeof(uint32_t) + sizeof(bool);
```

Comment:
what's the gain? https://stackoverflow.com/a/41132221/935325 says it's the same...